behaviour/graql/language/insert.feature

```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
+      define
+      residence sub relation,
+        relates resident,
+        relates place-of-residence,
+        has tenure-days,
+        key ref;
+      person plays resident;
+      address sub attribute, value string, plays place-of-residence;
+      tenure-days sub attribute, value long;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      insert
+      $p isa person, has name "Homer", has ref 0;
+      $addr "742 Evergreen Terrace" isa address;
+      $r (resident: $p, place-of-residence: $addr) isa residence, has ref 0;
+      """
+    When the integrity is validated
+    When get answers of graql query
+      """
+      match $td isa tenure-days; get;
+      """
+    Then answer size is: 0
+    Then graql insert
+      """
+      match
+        $r isa residence;
+      insert
+        $r has tenure-days 365;
+      """
+    Then get answers of graql query
+      """
+      match $r isa residence, has tenure-days $a; get $a;
+      """
+    Then concept identifiers are
+      |     | check | value           |
+      | RES | key   | ref:0           |
+      | TEN | value | tenure-days:365 |
+    Then uniquely identify answer concepts
+      | a   |
+      | TEN |
+
+
+  Scenario: an attribute ownership currently inferred by a rule can be explicitly inserted
+    Given graql define
+      """
+      define
+      lucy-is-aged-32 sub rule,
+      when {
+        $p isa person, has name "Lucy";
+      }, then {
+        $p has age 32;
+      };
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert $p isa person, has name "Lucy", has ref 0;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $p has age 32; get;
+      """
+    Then concept identifiers are
+      |      | check | value |
+      | LUCY | key   | ref:0 |
+    Then graql insert
+      """
+      match
+        $p has name "Lucy";
+      insert
+        $p has age 32;
+      """
+    Then the integrity is validated
+    Then get answers of graql query
+      """
+      match $p has age 32; get;
+      """
+    Then uniquely identify answer concepts
+      | p    |
+      | LUCY |
+
+
+  #############
+  # RELATIONS #
+  #############
+
+  Scenario: inserting a relation creates an instance of it
+    Given graql insert
+      """
+      insert
+      $p isa person, has ref 0;
+      $r (employee: $p) isa employment, has ref 1;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $r (employee: $p) isa employment; get;
+      """
+    Then concept identifiers are
+      |     | check | value |
+      | PER | key   | ref:0 |
+      | EMP | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | p   | r   |
+      | PER | EMP |
+
+
+  Scenario: when inserting a ternary relation that both owns an attribute and has an attribute as a roleplayer, both attributes are created
```

Comment:
ditch the `when`

Reply:
hmm actually maybe just a bit different:
"when inserting a relation that owns an attribute and has an attribute roleplayer, both attributes are created"

Reply:
Done!